plugins/SSH.koplugin/main.lua

```diff
@@ -0,0 +1,154 @@
+local Device =  require("device")
+local InfoMessage = require("ui/widget/infomessage")  -- luacheck:ignore
+local InputDialog = require("ui/widget/inputdialog")
+local UIManager = require("ui/uimanager")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local logger = require("logger")
+local util = require("util")
+local _ = require("gettext")
+
+--This plugin use a patched dropbear that add two things :
+--  the -n option to allow login without password
+--  read the keyfile from settings/SSH/authorized_keys
+
+if not util.pathExists("dropbearmulti") then return { disabled = true, } end
+
+local SSH = WidgetContainer:new{
+    name = 'SSH',
+    is_doc_only = false,
+}
+
+function SSH:init()
+    self.SSH_port = G_reader_settings:readSetting("SSH_port") or "2222"
+    self.allow_no_password = false
+    self.ui.menu:registerToMainMenu(self)
+end
+
+function SSH:start()
+    local cmd = string.format("%s %s %s %s %s %s",
+        "./dropbearmulti dropbear",
+        "-E", "-r settings/SSH/dropbear_rsa_host_key",
+        "-p", self.SSH_port,
+        "-P /tmp/dropbear.pid")
+     if self.allow_no_password then
+        cmd = string.format("%s %s", cmd, "-n")
+    end
+    if os.execute("command -v iptables") then
+        os.execute(string.format("%s %s %s",
+            "iptables -A INPUT -p tcp --dport", self.SSH_port,
+            "-m conntrack --ctstate NEW,ESTABLISHED -j ACCEPT"))
+        os.execute(string.format("%s %s %s",
+            "iptables -A OUTPUT -p tcp --sport", self.SSH_port,
+            "-m conntrack --ctstate ESTABLISHED -j ACCEPT"))
+    end
+    --An SSH/telnet server of course needs to be able to manipulate pseudoterminals...
```

Comment:
I feel like such a nitpicker saying things like this, but in the code we generally do `-- Comment starts here` :-)